File: cpp/src/transport/tls.cpp
/**
 * src/transport/tls.cpp
 */

#include "redfish_sdk/transport/tls.hpp"

namespace redfish {

TLSConfig build_tls_config(const ConnectionConfig& config) {
    TLSConfig tls;
    tls.verify      = config.verify_tls;
    tls.ca_cert     = config.tls_ca_cert;
    tls.client_cert = config.tls_client_cert;
    tls.client_key  = config.tls_client_key;
    return tls;
}

} // namespace redfish

File: python/redfish_sdk/transport/tls.py
"""
redfish_sdk/transport/tls.py

Builds the TLS config passed to httpx from ConnectionConfig.
Imports: models only.
"""

from __future__ import annotations

from redfish_sdk.models.redfish_types import ConnectionConfig, TLSConfig


def build_tls_config(config: ConnectionConfig) -> TLSConfig:
    """Map ConnectionConfig TLS fields to a TLSConfig for httpx."""
    if not config.verify_tls:
        return TLSConfig(verify=False)
    if config.tls_ca_cert:
        return TLSConfig(verify=config.tls_ca_cert)
    return TLSConfig(verify=True)
